plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -442,27 +467,32 @@ public String toString() {
 	
 	@Override
 	public <T extends IResource> List<T> getResources(String kind, String namespace) {
+		boolean needStrategy = initAuthorizationStrategy();
 		try {
-			if(client.getAuthorizationStrategy() == null) {
-				client.setAuthorizationStrategy(getAuthorizationStrategy());
-			}
-
 			client.setSSLCertificateCallback(OpenShiftCoreUIIntegration.getInstance().getSSLCertificateCallback());
 			return client.list(kind, namespace);
 		} catch (UnauthorizedException e) {
+			needStrategy = false;
 			return retryList("Unauthorized.  Trying to reauthenticate", e, kind, namespace);
+		} finally {
+			if(needStrategy) {
+				connect();
```

Comment:
this causes ConnectionTest#getResourceKindShouldCallClient() to fail with an NPE:
```
java.lang.NullPointerException
	at org.jboss.tools.openshift.core.connection.Connection.authorize(Connection.java:270)
	at org.jboss.tools.openshift.core.connection.Connection.connect(Connection.java:259)
	at org.jboss.tools.openshift.core.connection.Connection.getResources(Connection.java:482)
	at org.jboss.tools.openshift.core.connection.Connection.getResources(Connection.java:468)
	at org.jboss.tools.openshift.test.core.connection.ConnectionTest.getResourceKindShouldCallClient(ConnectionTest.java:97)
```